noir/noir-repo/compiler/noirc_evaluator/src/ssa/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -1757,6 +1757,30 @@ impl AcirContext {
         }
         Ok(())
     }
+
+    pub(crate) fn call_acir_function(
+        &mut self,
+        id: u32,
+        inputs: Vec<AcirValue>,
+        output_count: usize,
+    ) -> Result<Vec<AcirVar>, RuntimeError> {
+        let inputs = self.prepare_inputs_for_black_box_func_call(inputs)?;
+        let inputs = inputs
+            .iter()
+            .flat_map(|input| vecmap(input, |input| input.witness))
+            .collect::<Vec<_>>();
+        let outputs = vecmap(0..output_count, |_| self.acir_ir.next_witness_index());
```

Comment:
We can use `self.add_variable` to go straight to `AcirVar`s.

Reply:
I need both the witnesses (for setting up the opcode) and the vars (for constructing AcirValues). 
To use `add_variable` we would have to do something like such:
```rust
        let output_vars = vecmap(0..output_count, |_| self.add_variable());
        let output_witness = vecmap(output_vars, |var| self.var_to_witness(var).expect("ICE: should have just set variable"));
```
It feels a bit roundabout because we are inserting into the `AcirVars` map just to immediately fetch from it.

Reply:
Ah, I missed that. This way round makes sense then.